src/LiveComponent/tests/Functional/Form/ComponentWithFormTest.php

```diff
@@ -143,4 +144,105 @@ public function testFormRemembersValidationFromInitialForm(): void
             ->assertContains('The title field should not be blank')
         ;
     }
+
+    public function testHandleCheckboxChanges(): void
+    {
+        /** @var LiveComponentHydrator $hydrator */
+        $hydrator = self::getContainer()->get('ux.live_component.component_hydrator');
+
+        /** @var ComponentFactory $factory */
+        $factory = self::getContainer()->get('ux.twig_component.component_factory');
+
+        /** @var FormComponent1 $component */
+        $component = $factory->create('form_component1');
+
+        $dehydrated = $hydrator->dehydrate($component);
+        $bareForm = [
+            'text' => '',
+            'textarea' => '',
+            'range' => '',
+            'choice' => '',
+            'choice_expanded' => '',
+            'choice_multiple' => ['2'],
+            'select_multiple' => ['2'],
+            'checkbox_checked' => '1',
+            'file' => '',
+            'hidden' => '',
+        ];
```

Comment:
This test might be a bit easier to follow if:

A) In the form type, we do NOT use the `data` option on some fields to initialize them.
B) Instead, in the form component, add a `public $initialData` property (it does not need to be live)
C) Inside `instantiateForm()`, use `$this->initialData` as the 2nd arg when creating the form.

If you did this, then in this test, you could do `$component = $factory->create('form_component1', [ ... initial form data ])`. This would be more clear because it would be more obvious why we're asserting that, for example, `choice_multiple` should be equal to `['2']`.